vaultutil/pkg/fake/kube.go

```diff
@@ -0,0 +1,34 @@
+package fake
```

Comment:
Maybe it makes sense since this is testing-specific to put this in an `internal/fake` instead of `pkg/fake` so library consumers don't get confused? Or perhaps this particular file belongs in e.g. `k8sutil/pkg/test` which already has some helpers for the fake client?

Reply:
Yeah, I think the main reason I left it here originally was because it allows defining initial objects with the builder, so this could become highly customized per test suite. As well as possibly configuring specific schemes.

But I'd probably avoid that in general, and it's better to establish the default in `k8sutil` to start with.